File: javascript/src/max_stack.js
function processData(input) {
  stack = [];
  for (let i of input.split("\n").slice(1)) {
    let j = i
      .trim()
      .split(" ")
      .map(x => parseInt(x.trim()));
    const f = j[0];

    if (f === 1) {
      stack.push(j[1]);
    } else if (f === 2) {
      stack.pop();
    } else if (f === 3) {
      console.log(Math.max(...stack));
    }
  }
}

const input = `10
1 97
2
1 20
2
1 26
1 20
2
3
1 91
3`;

processData(input);

File: python/max_stack.py
def process_data(string=None):
    s = []
    for j in [list(map(lambda x: int(x), input().split(' '))) for c in range(int(input()))]:
        k = j[0]

        if k == 1:
            s.append(j[1])
        elif k == 2:
            s.pop()
        else:
            print(max(s))


process_data()


string = """10
1 97
2
1 20
2
1 26
1 20
2
3
1 91
3"""

process_data(string)
